db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/QueryTreeNodeVector.java

```diff
@@ -46,7 +48,22 @@
  * lists of query tree nodes inherit from QueryTreeNodeVector.
  */
 
-abstract class QueryTreeNodeVector<T extends QueryTreeNode> extends QueryTreeNode implements Iterable<T>{
+class QueryTreeNodeVector<T extends QueryTreeNode> extends QueryTreeNode implements Iterable<T>{
+
+    final Class<T> eltClass; // needed for cast in #acceptChildren
```

Comment:
What's the point of eltClass? The constructor that sets it does not seem to be used anywhere, and it looks kinda hacky.